### DISCUSSION TITLE
How do I save the checked/unchecked column names in the mtx-grid toolbar?

### DISCUSSION THREAD

**Discussion by rahulpd101** [ng-matero/ng-matero#655] [Q&A]
I am using the "showColumnMenuButton" property as true in mtx-grid to change the visible columns in the grid.
The issue faced by me is whenever the page is re-fetching the api to set the "columns" property, the unchecked columns automatically sets itself to checked.
I am not able to persist the column change choices.
Attaching screenshots for my table structure and web page:

**I used the (columnChange) event but found out that it also resets when the column is set again after subsquent api calls**

![front office column view](https://github.com/ng-matero/ng-matero/assets/37538741/9f7c95e3-f9c7-4133-a657-f5c824318aa8)
![mtx-grid code](https://github.com/ng-matero/ng-matero/assets/37538741/f9305d78-7a57-4f9d-a03c-8a342814b45e)

**Comment by nzbin** [comment]
I have fixed this issue in `17.1.3` and `16.3.3`, please try the new version.
